src/Promitor.Agents.Scraper/Scheduling/ResourcesScrapingJob.cs

```diff
@@ -285,13 +285,13 @@ private async Task ScrapeMetricBatched(BatchScrapeDefinition<IAzureResourceDefin
             try
             {
                 var resourceSubscriptionId = batchScrapeDefinition.ScrapeDefinitionBatchProperties.SubscriptionId;
-                var azureMonitorClient = _azureMonitorClientFactory.CreateIfNotExists(_metricsDeclaration.AzureMetadata.Cloud, _metricsDeclaration.AzureMetadata.TenantId,
+                var azureEnvironent = _metricsDeclaration.AzureMetadata.AzureEnvironment;
+                var azureMonitorClient = _azureMonitorClientFactory.CreateIfNotExists(azureEnvironent, _metricsDeclaration.AzureMetadata.TenantId,
                     resourceSubscriptionId, _metricSinkWriter, _azureScrapingSystemMetricsPublisher, _resourceMetricDefinitionMemoryCache, _configuration,
                     _azureMonitorIntegrationConfiguration, _azureMonitorLoggingConfiguration, _loggerFactory);
-                var azureEnvironent = _metricsDeclaration.AzureMetadata.Cloud.GetAzureEnvironment();
 
                 var tokenCredential = AzureAuthenticationFactory.GetTokenCredential(azureEnvironent.ManagementEndpoint, _metricsDeclaration.AzureMetadata.TenantId,
-                    AzureAuthenticationFactory.GetConfiguredAzureAuthentication(_configuration), new Uri(_metricsDeclaration.AzureMetadata.Cloud.GetAzureEnvironment().AuthenticationEndpoint));
+                    AzureAuthenticationFactory.GetConfiguredAzureAuthentication(_configuration), new Uri(_metricsDeclaration.AzureMetadata.AzureEnvironment.AuthenticationEndpoint));
```

Comment:
Probably we need a GetAzureEnvironment for AzureMetadata that has the logic to favor custom over cloud config